app/javascript/components/rooms/room/join/RoomJoin.jsx

```diff
@@ -100,15 +101,42 @@ export default function RoomJoin() {
     }
   }, [roomStatusAPI.isSuccess]);
 
+  // Plays a notify sound when the meeting starts if the user was in a waiting queue
+  const playNotifySound = () => {
+    const audio = new Audio(`data:audio/mpeg;base64,${NotifySound}`);
+    audio.play()
+      .catch((err) => {
+        console.error(err);
+      });
+  };
+
+  // Returns a random delay between 2 and 5 seconds, in increments of 250 ms
+  // The delay is to let the BBB server settle before attempting to join the meeting
+  // The randomness is to prevent multiple users from joining the meeting at the same time
+  const joinDelay = () => {
+    const min = 2000;
+    const max = 5000;
+    const step = 250;
+
+    // Calculate the number of possible steps within the given range
+    const numSteps = (max - min) / step;
+
+    // Generate a random integer from 0 to numSteps (inclusive)
+    const randomStep = Math.floor(Math.random() * (numSteps + 1));
+
+    // Calculate and return the random delay
+    return min + (randomStep * step);
+  };
+
   useEffect(() => {
     // Meeting started:
     //  When meeting starts this logic will be fired, indicating the event to waiting users (through a toast) for UX matter.
     //  Logging the event for debugging purposes and refetching the join logic with the user's given input (name & codes).
-    //  With a delay of 7s to give reasonable time for the meeting to fully start on the BBB server.
     if (hasStarted) {
       toast.success(t('toast.success.room.meeting_started'));
-      console.info(`Attempting to join the room(friendly_id): ${friendlyId} meeting in 7s.`);
-      setTimeout(methods.handleSubmit(handleJoin), 7000); // TODO: Amir - Improve this race condition handling by the backend.
+      playNotifySound();
+      console.info(`Attempting to join the room(friendly_id): ${friendlyId} meeting in 3s.`);
+      setTimeout(methods.handleSubmit(handleJoin), joinDelay()); // TODO: Amir - Improve this race condition handling by the backend.
```

Comment:
Can we move the sound to `handleJoin`? With this way, its possible that you hear the sound and just sit on the page waiting for 5 seconds

Reply:
Yes, of course.
It was very clunky with the 7 seconds, but since I've reduced the delay it felt pretty good.
Should we move the toast pop up in handleJoin as well?

Reply:
I feel like this is a case of perfect is enemy of the good.
It is not a perfect solution, but it is better than what we had.
And did not take much resources.

Reply:
The random delay can be done on the client side.
However, the issue now is the delay minimum that may cause users to experience issues on the auto join.
I think we'd better tackle that independently and improve the join workflow form the API itself.
Then having a delay that is less than 5 seconds for the sake of buffering will not cause any issues since the join URL will only be received after the meeting has actually became ready.

Reply:
Right. By moving the toast and sound into the delay it should be fine.

Reply:
Might have to do some tuning tho. I recall the Spinner stops at some point.